plugins/_getdir/listdir.php

```diff
@@ -0,0 +1,46 @@
+<?php
+require_once( '../../php/util.php' );
+require_once( '../../php/settings.php' );
+eval(FileUtil::getPluginConf("_getdir"));
+
+$theSettings = rTorrentSettings::get();
+
+$requestedDir = $_REQUEST['dir'];
+if(isset($requestedDir) && strlen($requestedDir))
+{
+	$dir = rawurldecode($requestedDir);
+	rTorrentSettings::get()->correctDirectory($dir);
+	$dir = FileUtil::addslash($dir);
+
+	if(
+			(strpos($dir,$topDirectory)!==0) ||
+			(($theSettings->uid>=0) && $checkUserPermissions && !Permission::doesUserHave($theSettings->uid,$theSettings->gid,$dir,0x0007))
+		)
+	{
+		$dir = $topDirectory;
+	}
+}
+else
+{
+	$dir = User::isLocalMode() ? $theSettings->directory : $topDirectory;
+	if(strpos(FileUtil::addslash($dir),$topDirectory)!==0)
+		$dir = $topDirectory;
+}
```

Comment:
Here is my revision to this code block to display the directory list properly by default. Please test and make any necessary changes. The file list was not displaying on my web client, due to missing a slash at the end.
```
else
{
	$dir = User::isLocalMode() ? $theSettings->directory : $topDirectory;
	$dir = FileUtil::addslash($dir);
	if(strpos($dir,$topDirectory)!==0)
		$dir = FileUtil::addslash($topDirectory);
}
```

Reply:
Is it because the `$topDirectory` config value lacking a trailing slash? The conf file says it explicitly that it needs a trailing slash:
```php
$topDirectory = '/';			// Upper available directory. Absolute path with trail slash.
```

I added an `addslash()` function to this variable, hope it can solve the problem.

Reply:
This did not solve my problem. I opted for a different fix instead.